lib/HLSL/HLOperationLower.cpp

```diff
@@ -8301,58 +8321,58 @@ void TranslateStructBufSubscriptUser(Instruction *user, Value *handle,
-        AtomicHelper helper(userCall, DXIL::OpCode::AtomicBinOp, handle, bufIdx,
-                            baseOffset);
+        AtomicHelper helper(userCall, DXIL::OpCode::AtomicBinOp, handle, 1,
+                            &bufIdx, baseOffset);
         TranslateAtomicBinaryOperation(helper, DXIL::AtomicBinOpCode::IMin,
                                        Builder, OP);
       } break;
       case IntrinsicOp::IOP_InterlockedUMax: {
-        AtomicHelper helper(userCall, DXIL::OpCode::AtomicBinOp, handle, bufIdx,
-                            baseOffset);
+        AtomicHelper helper(userCall, DXIL::OpCode::AtomicBinOp, handle, 1,
+                            &bufIdx, baseOffset);
         TranslateAtomicBinaryOperation(helper, DXIL::AtomicBinOpCode::UMax,
                                        Builder, OP);
       } break;
       case IntrinsicOp::IOP_InterlockedUMin: {
-        AtomicHelper helper(userCall, DXIL::OpCode::AtomicBinOp, handle, bufIdx,
-                            baseOffset);
+        AtomicHelper helper(userCall, DXIL::OpCode::AtomicBinOp, handle, 1,
+                            &bufIdx, baseOffset);
         TranslateAtomicBinaryOperation(helper, DXIL::AtomicBinOpCode::UMin,
                                        Builder, OP);
       } break;
       case IntrinsicOp::IOP_InterlockedOr: {
-        AtomicHelper helper(userCall, DXIL::OpCode::AtomicBinOp, handle, bufIdx,
-                            baseOffset);
+        AtomicHelper helper(userCall, DXIL::OpCode::AtomicBinOp, handle, 1,
+                            &bufIdx, baseOffset);
         TranslateAtomicBinaryOperation(helper, DXIL::AtomicBinOpCode::Or,
                                        Builder, OP);
       } break;
       case IntrinsicOp::IOP_InterlockedXor: {
-        AtomicHelper helper(userCall, DXIL::OpCode::AtomicBinOp, handle, bufIdx,
-                            baseOffset);
+        AtomicHelper helper(userCall, DXIL::OpCode::AtomicBinOp, handle, 1,
+                            &bufIdx, baseOffset);
         TranslateAtomicBinaryOperation(helper, DXIL::AtomicBinOpCode::Xor,
                                        Builder, OP);
       } break;
       case IntrinsicOp::IOP_InterlockedCompareStore:
       case IntrinsicOp::IOP_InterlockedCompareExchange: {
         AtomicHelper helper(userCall, DXIL::OpCode::AtomicCompareExchange,
-                            handle, bufIdx, baseOffset);
+                            handle, 1, &bufIdx, baseOffset);
         TranslateAtomicCmpXChg(helper, Builder, OP);
       } break;
       case IntrinsicOp::IOP_InterlockedCompareStoreFloatBitwise:
       case IntrinsicOp::IOP_InterlockedCompareExchangeFloatBitwise: {
         Type *i32Ty = Type::getInt32Ty(userCall->getContext());
         AtomicHelper helper(userCall, DXIL::OpCode::AtomicCompareExchange,
-                            handle, bufIdx, baseOffset, i32Ty);
+                            handle, 1, &bufIdx, baseOffset, i32Ty);
```

Comment:
Since these are all for structured buffers, they only have one coord and an offset that's handled separately. Should this approach be revived, I'll probably add more atomic testing.